server/pkg/api/http_test.go

```diff
@@ -870,6 +730,200 @@ func TestGameDrawReject(t *testing.T) {
 	}
 }
 
+func TestGamePlayerMakeMove(t *testing.T) {
+	player1, _, _, game1 := Setup(
+		[]setupOption{
+			WithPlayers(2),
+			WithPlayersInRoom(2),
+			WithRoom(),
+			WithGame(),
+		},
+	)
+
+	// Test making a valid move.
+	testcases := []struct {
+		description              string
+		id                       string
+		body                     string
+		expectedResponseContains []string
+		expectedStatusCode       int
+	}{
+		{
+			"Valid gameID and move.",
+			game1.GetID().String(),
+			fmt.Sprintf("{\"player_id\":\"%s\",\"move\":{\"source\":{\"rank\":1,\"file\":1},\"destination\":{\"rank\":2,\"file\":1},\"move_type\":\"NORMAL\"}}", player1.GetID()),
+			[]string{
+				"\"winning_players\":[]",
+				"\"losing_players\":[]",
+				"\"drawn_players\":[]",
+			},
+			200,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.description, func(t *testing.T) {
+			router := &mux.Router{}
+			AttachRoutes(router)
+
+			request, _ := http.NewRequest(
+				"POST",
+				fmt.Sprintf("/api/game/%s/move", tc.id),
+				strings.NewReader(tc.body),
+			)
+			writer := executeRequest(router, request)
+
+			assert.Equal(t, tc.expectedStatusCode, writer.statusCode)
+			responseString := string(writer.response)
+			for _, e := range tc.expectedResponseContains {
+				assert.Contains(t, responseString, e)
+			}
+		})
+	}
+}
+
+type setupOption = func(s *setupBuilder) *setupBuilder
+
+type setupBuilder struct {
+	player1          *player.Player
+	player2          *player.Player
+	addPlayer1InRoom bool
+	addPlayer2InRoom bool
+	room1            *room.Room
+	game1            *game.Game
+}
+
+func (s *setupBuilder) Build() (*player.Player, *player.Player, *room.Room, *game.Game) {
+	return s.player1, s.player2, s.room1, s.game1
+}
+
+func Setup(options []setupOption) (*player.Player, *player.Player, *room.Room, *game.Game) {
```

Comment:
```suggestion
func Setup(options ...setupOption) (*player.Player, *player.Player, *room.Room, *game.Game) {
```

this would make it cleaner on the caller and encapsulate type

Reply:
what if we create something like

```
type testEntites struct {
  *player.Player, 
  *player.Player, 
  *room.Room, 
  *game.Game
}
```

just food for thought, would encapsulate return values and make it extensible by just adding another prop to struct

Reply:
Done